cdap-metrics-writer-ext-gcp-monitoring/src/main/java/io/cdap/cdap/metrics/process/gcp/CloudMonitoringWriter.java

```diff
@@ -137,6 +138,20 @@ public void initialize(MetricsWriterContext metricsWriterContext) {
   }
 
   private void populateAutoFilledMap(Map<String, String> properties, String platformVersion) {
+    if (platformVersion == null || platformVersion.isEmpty()) {
```

Comment:
nit + optional: Maybe we can improve the logic to throw an error with all missing fields instead of one at a time? That prevents the user from fixing one issue then running into another.

Reply:
Done